### PR TITLE
[PyTorch] move mask types to fprop

### PR DESCRIPTION
**Pull request by ksivaman**

### CODE REVIEW

File: transformer_engine/pytorch/softmax.py [hunk 1]
```diff
@@ -257,6 +255,7 @@ def forward(
         self.input_in_fp16 = inp.dtype == torch.float16
         self.input_in_bf16 = inp.dtype == torch.bfloat16
         self.input_in_float16 = self.input_in_fp16 or self.input_in_bf16
+        self.attn_mask_type = attn_mask_type
```
Comment (timmoon10) [inline transformer_engine/pytorch/softmax.py:258]: I suppose this is fine as a quick hack, but it would be better to pass it as an argument to `is_kernel_available`, `forward_fused_softmax`, and `forward_torch_softmax` instead of treating it as a member variable.

File: tests/pytorch/test_onnx_export.py [hunk 2]
```diff
@@ -1193,9 +1190,9 @@ def test_export_core_attention(
 
 test_configs_multihead_attention = [
     #"use_mask, attn_mask_type"
-    (False,    "causal"),  # calls ScaledUpperTriangMaskedSoftmax
+    (False,    "no_mask"), # calls ScaledUpperTriangMaskedSoftmax
```
Comment (timmoon10) [inline tests/pytorch/test_onnx_export.py:1193]: We're repeating a test:
```suggestion
    (False,    "causal"),  # calls ScaledUpperTriangMaskedSoftmax
```

**Review by timmoon10 — APPROVED** [review]
Overall LGTM.

While we're tweaking the API, it might be worthwhile thinking about changing the default mask type to "no_mask" to match the PyTorch default (https://pytorch.org/docs/stable/generated/torch.nn.Transformer.html#torch.nn.Transformer.forward). Also, "causal" doesn't make sense for visual Transformers. Probably better to do it after this release though.

File: tests/pytorch/test_onnx_export.py (hunk 2 above)
Comment (timmoon10) [reply tests/pytorch/test_onnx_export.py:1193]: Minor comment fix:
```suggestion
(False, "no_mask"),  # calls ScaledSoftmax
```

**Comment by nzmora-nvidia** [comment]
To export to ONNX we trace the graph. This trace cannot capture dynamic control flow.
This MR breaks export to ONNX
